File: api/gateway/native/src/types/response/book.ts
export interface IBookResponse {
  id: number
  title: string
  titleKana: string
  description: string
  isbn: string
  publisher: string
  publishedOn: string
  thumbnailUrl: string
  rakutenUrl: string
  size: string
  author: string
  authorKana: string
  createdAt: string
  updatedAt: string
  bookshelf?: IBookResponseBookshelf
}

export interface IBookResponseBookshelf {
  id: number
  status: string
  impression: string
  readOn: string
  createdAt: string
  updatedAt: string
}

export interface IReviewResponse {
  id: number
  impression: string
  createdAt: string
  updatedAt: string
  book: IReviewResponseBook
  user: IReviewResponseUser
}

export interface IReviewResponseBook {
  id: number
  title: string
  thumbnailUrl: string
}

export interface IReviewResponseUser {
  id: string
  username: string
  thumbnailUrl: string
}

export interface IUserReviewListResponse {
  reviews: Array<IUserReviewListResponseReview>
  limit: number
  offset: number
  total: number
  order?: IUserReviewListResponseOrder
}

export interface IUserReviewListResponseReview {
  id: number
  impression: string
  createdAt: string
  updatedAt: string
  book: IUserReviewListResponseBook
}

export interface IUserReviewListResponseBook {
  id: number
  title: string
  thumbnailUrl: string
}

export interface IUserReviewListResponseOrder {
  by: string
  direction: string
}

export interface IBookReviewListResponse {
  reviews: Array<IBookReviewListResponseReview>
  limit: number
  offset: number
  total: number
  order?: IBookReviewListResponseOrder
}

export interface IBookReviewListResponseReview {
  id: number
  impression: string
  createdAt: string
  updatedAt: string
  user: IBookReviewListResponseUser
}

export interface IBookReviewListResponseUser {
  id: string
  username: string
  thumbnailUrl: string
}

export interface IBookReviewListResponseOrder {
  by: string
  direction: string
}

File: api/gateway/native/internal/response/v1/book.go
package v1

// 書籍情報
type BookResponse struct {
	ID           int64          `json:"id"`           // 書籍ID
	Title        string         `json:"title"`        // タイトル
	TitleKana    string         `json:"titleKana"`    // タイトル(かな)
	Description  string         `json:"description"`  // 説明
	Isbn         string         `json:"isbn"`         // ISBN
	Publisher    string         `json:"publisher"`    // 出版社名
	PublishedOn  string         `json:"published_on"` // 出版日
	ThumbnailURL string         `json:"thumbnailUrl"` // サムネイルURL
	RakutenURL   string         `json:"rakutenUrl"`   // 楽天ショップURL
	Size         string         `json:"size"`         // 楽天書籍サイズ
	Author       string         `json:"author"`       // 著者名一覧
	AuthorKana   string         `json:"authorKana"`   // 著者名一覧(かな)
	Bookshelf    *BookBookshelf `json:"bookshelf"`    // ユーザーの本棚情報
	CreatedAt    string         `json:"createdAt"`    // 登録日時
	UpdatedAt    string         `json:"updatedAt"`    // 更新日時
}

type BookBookshelf struct {
	ID         int64  `json:"id"`         // 本棚ID
	Status     string `json:"status"`     // 読書ステータス
	ReadOn     string `json:"readOn"`     // 読み終えた日
	Impression string `json:"impression"` // 感想
	CreatedAt  string `json:"createdAt"`  // 登録日時
	UpdatedAt  string `json:"updatedAt"`  // 更新日時
}
